app/models/activity.rb

```diff
@@ -602,6 +603,22 @@ def historic?
     programme_status.in?(["completed", "stopped", "cancelled"])
   end
 
+  def ensure_linked_activity_reciprocity
+    return unless saved_change_to_attribute?(:linked_activity_id)
+
+    formerly_linked_activity_id = saved_change_to_attribute(:linked_activity_id).first
+    if formerly_linked_activity_id && (formerly_linked_activity = Activity.find_by(id: formerly_linked_activity_id)).present?
+      # we do not want to propagate this, because it would nullify self.linked_activity_id
```

Comment:
> What do we mean by `propagate` here? Removing the linked activity from `self` as well?

We mean "make the formerly linked activity invoke the post-save callback". That's why we're calling `update_columns`, which doesn't trigger model callbacks. (If we allowed it to trigger callbacks, it would remove the `linked_activity_id` from the current activity, because for the formerly linked activity, that's the `formerly_linked_activity`, so it would cancel out our efforts to link it to its new "link partner".

Reply:
> Just for my own understanding, am I right in reading that this method is:
> 
> 1. Removing any formerly linked activities
> 2. Making sure the newly-linked activity's `linked_activity` is set to the current activity
> 
> ?

Yep, that's what it's doing. It made sense to us to have one method, I could be persuaded to change it but only if there was a very strong argument. >.>

Reply:
Ok, we've agreed to leave this for now and possibly address in a future commit